projects/ngneat/helipopper/src/lib/utils.ts

```diff
@@ -47,42 +47,38 @@ export function inView(
   });
 }
 
-function isElementOverflow(host: HTMLElement) {
-  const parentEl = host.parentElement;
-  const parentTest = host.offsetWidth > parentEl.offsetWidth;
-  const elementTest = host.offsetWidth < host.scrollWidth;
-
-  return parentTest || elementTest;
+function isElementOverflow(host: HTMLElement): boolean {
+  // Don't access the `offsetWidth` multipe times since it triggers layout updates.
+  const hostOffsetWidth = host.offsetWidth;
+  return hostOffsetWidth > host.parentElement.offsetWidth || hostOffsetWidth < host.scrollWidth;
 }
 
-export function overflowChanges(host: TippyElement) {
+export function overflowChanges(ngZone: NgZone, host: TippyElement) {
   const element = coerceElement(host);
 
-  return dimensionsChanges(element).pipe(
+  return dimensionsChanges(ngZone, element).pipe(
     auditTime(150),
-    map(() => {
-      return isElementOverflow(element);
-    })
+    map(() => isElementOverflow(element))
   );
 }
 
-export function dimensionsChanges(target: HTMLElement) {
-  return resizeObserverStrategy(target);
+export function dimensionsChanges(ngZone: NgZone, target: HTMLElement) {
```

Comment:
These functions are [public](https://github.com/ngneat/helipopper/blob/master/projects/ngneat/helipopper/src/public-api.ts#L5).

It should consider as a breaking change.

Reply:
Oh, I haven't seen them to be exposed. We already updated `inView` function in the previous PR. Should I leave the change as it is or move `runOutsideAngular` to directive?

Reply:
Yes, please. I think it doesn't worth the major version.

Reply:
Hey, I've removed `ngZone` argument from `inView` and resize observer, moved just the whole code into `runOutsideAngular` inside the `ngAfterViewInit`.

Reply:
Thanks